### PR TITLE
fix: nested refs

### PR DESCRIPTION
**Pull request by peter-mouland**
Start work towards https://github.com/fastify/fastify-swagger/issues/462

TBC:
 - [ ] how to handle ids/refs that arent modified using `buildLocalReference`?
     - Current behaviour `t.equal(definitions['def-3'].properties.b.$ref, '#/definitions/schemaB')`, vs
     - more correct `t.equal(definitions['def-3'].properties.b.$ref, '#/definitions/def-2')`?
 - [ ] de we need to code for `key === 'examples'` in the swagger spec?

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/spec/swagger/utils.js [hunk 1]
```diff
@@ -266,9 +266,28 @@ function prepareSwaggerMethod (schema, ref, swaggerObject) {
   return swaggerMethod
 }
 
+function transformDefs (jsonSchema) {
+  if (typeof jsonSchema === 'object' && jsonSchema !== null) {
+    Object.keys(jsonSchema).forEach(function (key) {
+      if (key === '$ref') {
+        jsonSchema[key] = `#/definitions/${jsonSchema[key]}`
+      } else if (key === 'examples' && Array.isArray(jsonSchema[key]) && (jsonSchema[key].length > 1)) {
+        jsonSchema.examples = convertExamplesArrayToObject(jsonSchema.examples)
```
Comment (Eomm) [inline lib/spec/swagger/utils.js:275]: Is this case tested?

Comment (peter-mouland) [reply lib/spec/swagger/utils.js:275]: not at the moment, I'm not sure why the code is handing `examples`.  Should it be tested, or removed?

Comment (Eomm) [reply lib/spec/swagger/utils.js:275]: We landed the https://github.com/fastify/fastify-swagger/pull/472 PR

is this needed?

Comment (peter-mouland) [reply lib/spec/swagger/utils.js:275]: I'd say, this PR is still needed as this PR takes care of both Swagger and OpenAPI,  that otehr PR only takes care of OpenAPI?

Does another test need to be added for when the user does not tweak `buildLocalReference`, or is returning the $id now preferred?

Comment (peter-mouland) [reply lib/spec/swagger/utils.js:275]: closed in favour of https://github.com/fastify/fastify-swagger/pull/479

**Comment by peter-mouland** [comment]
closed in favour of https://github.com/fastify/fastify-swagger/pull/479

**Pull request closed without merging**